packages/effect/src/Duration.ts

```diff
@@ -515,6 +515,59 @@ export const clamp: {
     })
 )
 
+/**
+ * @since 2.4.19
+ * @category math
+ */
+export const divide: {
+  (by: number): (self: DurationInput) => Option.Option<Duration>
+  (self: DurationInput, by: number): Option.Option<Duration>
+} = dual(
+  2,
+  (self: DurationInput, by: number): Option.Option<Duration> =>
+    match(self, {
```

Comment:
You can simplify this by wrapping unsafeDivide with a try catch.

Reply:
The results are a bit different (https://github.com/Effect-TS/effect/pull/2494#issuecomment-2047454353).